### PR TITLE
[NU-1791] Add varargs to concat and merge helper functions

### PR DESCRIPTION
**Pull request by lukasz-bigorajski**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala [hunk 1]
```diff
@@ -27,25 +29,31 @@ trait CollectionUtils extends HideToString {
...
-    "Merges two maps. Values in the first map will be overwritten with values from the second map if keys are the same"
+    "Merges maps. Values in the first map will be overwritten with values from the other one if keys are the same"
   )
-  @GenericType(typingFunction = classOf[MapMergeTyping])
+  @GenericType(typingFunction = classOf[VarargsMapMergeTyping])
+  @varargs
   def merge[K, V](
       @ParamName("map1") map1: java.util.Map[K, V],
-      @ParamName("map2") map2: java.util.Map[K, V]
+      @ParamName("map2") map2: java.util.Map[K, V]*
```
Comment (lciolecki) [inline utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:52]: Why not: 
```
@ParamName("maps") maps: java.util.Map[K, V]*
```
???

Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:52]: Done

Comment (lciolecki) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:52]: I mean do we need two params? Can't we use only vargs:

```
@ParamName("maps") maps: java.util.Map[K, V]*
```
??

Comment (lciolecki) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala:52]: The same with concat helper

Comment (lukasz-bigorajski) [reply utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala]: IMHO it is a simple way to ensure that user pass at least one collection ;)

**Review by lciolecki — APPROVED** [review]
Good job ;)

**Pull request merged**